### ISSUE TITLE
Issue while creating a ticket

### ISSUE THREAD

**Issue by andrewrussobaca** [SalesforceLabs/IssueBox#6]
This error occurs when a user tries to create a new issue. I am not sure what is causing this...

An error occurred while trying to update the record. Please try again.
We can't save this record because the “IssueBox” process failed. Give your Salesforce admin these details. afl.afl_InsertArticleFeedback: execution of AfterInsert caused by: System.NullPointerException: Attempt to de-reference a null object Class.afl.afl_ArticleFeedbackSecurityHandler.isCreateable: line 113, column 1 Class.afl.afl_ArticleFeedbackSecurityHandler.isCreateable: line 182, column 1 Trigger.afl.afl_InsertArticleFeedback: line 8, column 1 Error ID: 8514839-217726 (-1341328723)

**Comment by jrattanpal** [comment]
Can you share the debug log with me?

And can you confirm you have assigned proper permission sets to the user?

And I am not sure what afl.afl_InsertArticleFeedback is. THat's not
IssueBox trigger or class

**Comment by andrewrussobaca** [comment]
Yep. I think the issue is some conflict with this app exchange package. 

https://appexchange.salesforce.com/appxListingDetail?listingId=a0N3A00000FP1A8UAL <https://appexchange.salesforce.com/appxListingDetail?listingId=a0N3A00000FP1A8UAL>


Here is the log
48.0 APEX_CODE,FINEST;APEX_PROFILING,INFO;CALLOUT,INFO;DB,INFO;NBA,INFO;SYSTEM,DEBUG;VALIDATION,INFO;VISUALFORCE,INFO;WAVE,INFO;WORKFLOW,INFO
11:11:16.0 (182840)|USER_INFO|[EXTERNAL]|0051M000007feG0|csearle@bacasystems.com|(GMT-04:00) Eastern Daylight Time (America/New_York)|GMT-04:00
11:11:16.0 (252245)|EXECUTION_STARTED
11:11:16.0 (257656)|CODE_UNIT_STARTED|[EXTERNAL]|Flow:01I1M000002cT0s
11:11:16.0 (584525)|CODE_UNIT_FINISHED|Flow:01I1M000002cT0s
11:11:16.0 (593548)|EXECUTION_FINISHED
11:11:16.5 (5706684)|USER_INFO|[EXTERNAL]|0051M000007feG0|csearle@bacasystems.com|(GMT-04:00) Eastern Daylight Time (America/New_York)|GMT-04:00
11:11:16.5 (8890877)|ENTERING_MANAGED_PKG|issuebox
11:11:16.11 (11513981)|CUMULATIVE_LIMIT_USAGE
11:11:16.11 (11513981)|LIMIT_USAGE_FOR_NS|(default)|
  Number of SOQL queries: 0 out of 100
  Number of query rows: 0 out of 50000
  Number of SOSL queries: 0 out of 20
  Number of DML statements: 0 out of 150
  Number of DML rows: 0 out of 10000
  Maximum CPU time: 0 out of 10000
  Maximum heap size: 0 out of 6000000
  Number of callouts: 0 out of 100
  Number of Email Invocations: 0 out of 10
  Number of future calls: 0 out of 50
  Number of queueable jobs added to the queue: 0 out of 50
  Number of Mobile Apex push calls: 0 out of 10

11:11:16.11 (11513981)|CUMULATIVE_LIMIT_USAGE_END

11:11:16.45 (45811137)|CODE_UNIT_STARTED|[EXTERNAL]|Workflow:01I1M000002cT0s
11:11:16.45 (56020302)|WF_RULE_EVAL_BEGIN|Workflow
11:11:16.45 (56065693)|WF_CRITERIA_BEGIN|[Issue: I-0012 a4t1M000000LIrl]|Issue_new_issue_notification3011M00000010H0|01Q1M000000bo66|ON_CREATE_ONLY|0
11:11:16.45 (56243587)|WF_FORMULA|Formula:ENCODED:[treatNullAsNull]true|Values:
11:11:16.45 (56252913)|WF_CRITERIA_END|true
11:11:16.45 (56422472)|WF_CRITERIA_BEGIN|[Issue: I-0012 a4t1M000000LIrl]|issuebox_IssueBox3011M00000010EL|01Q1M000000boDq|ON_ALL_CHANGES|0
11:11:16.45 (56501713)|WF_FORMULA|Formula:ENCODED:[treatNullAsNull]true|Values:
11:11:16.45 (56510268)|WF_CRITERIA_END|true
11:11:16.45 (56530031)|WF_SPOOL_ACTION_BEGIN|Workflow
11:11:16.45 (56572392)|WF_ACTION| Flow Trigger: 2;
11:11:16.45 (56586683)|WF_RULE_EVAL_END
11:11:16.58 (58002390)|FLOW_CREATE_INTERVIEW_BEGIN|00DG0000000lSeW|3001M000000U3JD|3011M00000010H0
11:11:16.58 (65515947)|FLOW_CREATE_INTERVIEW_END|7318e3b028d473b25e9146bfed78172285666ca-77e2|Issue: new issue notification
11:11:16.45 (65608621)|WF_FLOW_ACTION_BEGIN|09L1M000000H017
11:11:16.66 (66305585)|FLOW_START_INTERVIEWS_BEGIN|1
11:11:16.66 (66364730)|FLOW_START_INTERVIEW_BEGIN|7318e3b028d473b25e9146bfed78172285666ca-77e2|Issue: new issue notification
11:11:16.66 (66701467)|FLOW_START_INTERVIEW_END|7318e3b028d473b25e9146bfed78172285666ca-77e2|Issue: new issue notification
11:11:16.66 (224067141)|WF_EMAIL_SENT|Reference:issuebox__Issue__c.Issue_Creation_alert|Recipients:arusso@bacasystems.com |CcEmails:
11:11:16.66 (224873262)|FLOW_START_INTERVIEWS_END|1
11:11:16.66 (224932757)|FLOW_INTERVIEW_FINISHED|7318e3b028d473b25e9146bfed78172285666ca-77e2|Issue: new issue notification
11:11:16.45 (226421841)|WF_FLOW_ACTION_END|09L1M000000H017
11:11:16.227 (227727941)|FLOW_CREATE_INTERVIEW_BEGIN|00DG0000000lSeW|3001M000000U3J3|3011M00000010EL
11:11:16.227 (232215279)|FLOW_CREATE_INTERVIEW_END|7525e3b028d473b25e9146bfed78172285666ca-7713|IssueBox
11:11:16.45 (232288091)|WF_FLOW_ACTION_BEGIN|09L1M000000H07Z
11:11:16.232 (232743406)|FLOW_START_INTERVIEWS_BEGIN|1
11:11:16.232 (232801572)|FLOW_START_INTERVIEW_BEGIN|7525e3b028d473b25e9146bfed78172285666ca-7713|IssueBox
11:11:16.232 (233083984)|FLOW_START_INTERVIEW_END|7525e3b028d473b25e9146bfed78172285666ca-7713|IssueBox
11:11:16.239 (239619617)|CODE_UNIT_STARTED|[EXTERNAL]|Flow:FeedItem
11:11:16.239 (240004112)|CODE_UNIT_FINISHED|Flow:FeedItem
11:11:16.293 (293724819)|ENTERING_MANAGED_PKG|afl
11:11:16.293 (294319676)|ENTERING_MANAGED_PKG|afl
11:11:16.293 (294366968)|ENTERING_MANAGED_PKG|afl
11:11:16.293 (695389256)|ENTERING_MANAGED_PKG|afl
11:11:17.126 (1126764701)|CUMULATIVE_LIMIT_USAGE
11:11:17.126 (1126764701)|LIMIT_USAGE_FOR_NS|(default)|
  Number of SOQL queries: 0 out of 100
  Number of query rows: 0 out of 50000
  Number of SOSL queries: 0 out of 20
  Number of DML statements: 0 out of 150
  Number of DML rows: 0 out of 10000
  Maximum CPU time: 0 out of 10000
  Maximum heap size: 0 out of 6000000
  Number of callouts: 0 out of 100
  Number of Email Invocations: 0 out of 10
  Number of future calls: 0 out of 50
  Number of queueable jobs added to the queue: 0 out of 50
  Number of Mobile Apex push calls: 0 out of 10

11:11:17.126 (1126764701)|CUMULATIVE_LIMIT_USAGE_END

11:11:17.162 (1162686311)|CODE_UNIT_STARTED|[EXTERNAL]|Flow:FeedItem
11:11:17.162 (1162749265)|CODE_UNIT_FINISHED|Flow:FeedItem
11:11:17.204 (1204345533)|ENTERING_MANAGED_PKG|afl
11:11:17.204 (1204624889)|ENTERING_MANAGED_PKG|afl
11:11:17.204 (1204763641)|ENTERING_MANAGED_PKG|afl
11:11:17.204 (1204833138)|ENTERING_MANAGED_PKG|afl
11:11:17.205 (1205282670)|CUMULATIVE_LIMIT_USAGE
11:11:17.205 (1205282670)|LIMIT_USAGE_FOR_NS|(default)|
  Number of SOQL queries: 0 out of 100
  Number of query rows: 0 out of 50000
  Number of SOSL queries: 0 out of 20
  Number of DML statements: 0 out of 150
  Number of DML rows: 0 out of 10000
  Maximum CPU time: 0 out of 10000
  Maximum heap size: 0 out of 6000000
  Number of callouts: 0 out of 100
  Number of Email Invocations: 0 out of 10
  Number of future calls: 0 out of 50
  Number of queueable jobs added to the queue: 0 out of 50
  Number of Mobile Apex push calls: 0 out of 10

11:11:17.205 (1205282670)|CUMULATIVE_LIMIT_USAGE_END

11:11:17.233 (1233285805)|FLOW_ELEMENT_ERROR|afl.afl_InsertArticleFeedback: execution of AfterInsert

caused by: System.NullPointerException: Attempt to de-reference a null object

(afl)


|FlowActionCall|myRule_4_A1
11:11:17.233 (1267490236)|FLOW_START_INTERVIEWS_END|1
11:11:17.268 (1268021795)|WF_FLOW_ACTION_ERROR|09L1M000000H07Z|3001M000000U3J3|Error executing flow: issuebox__IssueBox, FlowDefId: 3001M000000U3J3, FlowVersionId: 3011M00000010EL
11:11:17.268 (1268088059)|WF_FLOW_ACTION_ERROR_DETAIL|afl.afl_InsertArticleFeedback: execution of AfterInsert

caused by: System.NullPointerException: Attempt to de-reference a null object

(afl)



11:11:17.268 (1268189140)|WF_FLOW_ACTION_END|09L1M000000H07Z

**Comment by andrewrussobaca** [comment]
I uninstalled the article feedback package and the issue box package stated to work for that user without an issue... I think there was a conflict with it.

**Comment by jrattanpal** [comment]
That is very strange as it's a namespaced package and shouldn't have any
issue there. We don't touch anything related to feedback so it shouldn't
cause that issue either.

Did you make any triggered/Flows/Processes that change Feedback object when
issue is created?

afl.afl_InsertArticleFeedback shouldn't be triggered at all.

**Comment by andrewrussobaca** [comment]
Nope. I the only thing changed with the package was I added the article feedback in the lighting page layout of the article page. The error with issue box happens on other record pages unrelated to articles

**Comment by jrattanpal** [comment]
That is so strange. I may need to test it (in a project this week) sometime
and see if I can figure out what happened here.

For now, I think you are unlocked. RIght?

**Comment by andrewrussobaca** [comment]
Yep
